src/server/db_slice.cc

```diff
@@ -468,6 +468,7 @@ OpResult<DbSlice::PrimeItAndExp> DbSlice::FindInternal(const Context& cntx, std:
       db.prime.CVCUponBump(change_cb_.back().first, res.it, bump_cb);
     }
 
+    block_counter_.Wait();  // We must not change the bucket's internal order during serialization
```

Comment:
I would have to moved it to line 463 such that the whole block (including the capture version change) is atomic but it should be fine this way as well